File: StockControl-UI/src/types/transaction.ts
export enum TransactionType {
  CheckIn,
  CheckOut,
}

export interface Transaction {
  id: string;
  productCode: string;
  quantity: number;
  type: TransactionType;
  date: Date;
}

export interface AddTransactionPayload {
  productCode: string;
  type: TransactionType;
  quantity: number;
}

File: StockControl/src/Domain/Entities/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StockControl.Domain.Entities
{
    public class Transaction
    {
        public int Id { get; set; }
        public Product Product { get; set; }
        public TransactionType Type { get; set; }
        public DateTime CreatedAt { get; set; }
        public int Quantity { get; set; }
    }

    public enum TransactionType
    {
        Checkin,
        Checkout
    }
}
